kube-controller-manager: convert to structured logging

pohly opened the pull request:
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Most of the individual controllers were already converted earlier. Adding of the name to the logger used by each controller gets consolidated in this commit. By using the name under which the controller is registered we ensure that the names in the log are consistent.

#### Which issue(s) this PR fixes:
Related-to #https://github.com/kubernetes/kubernetes/pull/113464

#### Special notes for your reviewer:

This PR should get merged into 1.27 after the other PRs which change controllers.

TODO:
- [x] remove `LoggerWithName` calls from individual controllers
- [x] update `hack/logcheck.conf`

#### Does this PR introduce a user-facing change?
```release-note
Migrated the main kube-controller-manager binary to use [contextual logging](https://k8s.io/docs/concepts/cluster-administration/system-logs/#contextual-logging).
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/3077-contextual-logging
```

pohly committed:
kube-controller-manager: convert to structured logging

Most of the individual controllers were already converted earlier. Some log
calls were missed or added and then not updated during a rebase. Some of those
get updated here to fill those gaps.

Adding of the name to the logger used by each controller gets
consolidated in this commit. By using the name under which the
controller is registered we ensure that the names in the log
are consistent.

mengjiao-liu commented on cmd/kube-controller-manager/app/plugins_providers.go:
```diff
@@ -60,7 +61,7 @@ type pluginInfo struct {
 }
 
-func appendAttachableLegacyProviderVolumes(allPlugins []volume.VolumePlugin, featureGate featuregate.FeatureGate) ([]volume.VolumePlugin, error) {
+func appendAttachableLegacyProviderVolumes(logger klog.Logger, allPlugins []volume.VolumePlugin, featureGate featuregate.FeatureGate) ([]volume.VolumePlugin, error) {
```
CI failed: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/116529/pull-kubernetes-typecheck/1635202025572536320
> FAILED   verify-typecheck-providerless.sh	108s

This function in file `cmd/kube-controller-manager/app/plugins_providerless.go` also needs to add the parameter logger `klog.Logger`

pohly replied:
Ah, that file has a `+build providerless` - that's why it's not usually built and I didn't see that failure earlier. `hack/verify-golangci-lint.sh` also doesn't find it. Interesting  - I wonder whether it should.

But that's an aside. I've fixed it locally and will push the fix in the next update.

pohly committed:
hack/logcheck.conf: enforce contextual logging in kube-controller-manager

Not all PRs made it into 1.27, but enough that it is now easier to list the
exceptions.

soltysh reviewed (approved):
/lgtm
/approve
/milestone v1.27

soltysh commented:
/triage accepted
/priority important-longterm
